hackathon_site/hardware/test_api.py

```diff
@@ -1708,3 +1743,108 @@ def test_failed_beginning_status(self):
+            username="frank@johnston.com",
+            password="hellothere31415",
+            email="frank@johnston.com",
+            first_name="Frank",
+            last_name="Johnston",
+        )
+        self.user3 = User.objects.create_user(
+            username="franklin@carmichael.com",
+            password="supersecret456",
+            email="franklin@carmichael.com",
+            first_name="Franklin",
+            last_name="Carmichael",
+        )
+        self.user4 = User.objects.create_user(
+            username="lawren@harris.com",
+            password="wxyz7890",
+            email="lawren@harris.com",
+            first_name="Lawren",
+            last_name="Harris",
+        )
+
+        self.user5 = User.objects.create_user(
+            username="law@henn.com",
+            password="wxyzdfs890",
+            email="law@henn.com",
+            first_name="Law",
+            last_name="Henn",
+        )
+
+        self.hardware1 = Hardware.objects.create(
+            name="aHardware",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=10,
+            max_per_team=5,
+            picture="/picture/location",
+        )
+
+        self.order = Order.objects.create(
+            status="Cart",
+            team=self.team,
+            request={"hardware": [{"id": 1, "quantity": 2},]},
+        )
+
+        self.category1 = Category.objects.create(name="category1", max_per_team=4)
+
+        self.hardware1.categories.add(self.category1)
+
+        self.view = reverse("api:hardware:order-list")
+
+    def test_team_less_min_order(self):
+        self._login()
+
+        request_data = {"hardware": [{"id": self.hardware1.id, "quantity": 2}]}
+        response = self.client.post(self.view, request_data, format="json")
+        self.assertEqual(
+            response.json(),
+            {"non_field_errors": ["Does not match team size criteria"]},
+        )
+
+    def test_team_more_max_order(self):
+        self._login()
+        # for user in self._create_user_set():
+        #     if self._create_user_set().index(user) != 0:
+        #         self.profile = Profile.objects.create(user=user, team=self.team)
+
+        self.profile2 = Profile.objects.create(user=self.user2, team=self.team)
+        self.profile3 = Profile.objects.create(user=self.user3, team=self.team)
+        self.profile4 = Profile.objects.create(user=self.user4, team=self.team)
+        self.profile5 = Profile.objects.create(user=self.user5, team=self.team)
+
+        request_data = {"hardware": [{"id": self.hardware1.id, "quantity": 2}]}
+        response = self.client.post(self.view, request_data, format="json")
+        self.assertEqual(
+            response.json(),
+            {"non_field_errors": ["Does not match team size criteria"]},
+        )
+
+    def test_team_correct_size_order(self):
```

Comment:
you can remove this test because almost all the tests in OrderListViewPostTestCase class test the correct team order, you only need to test if the constraints are not met (your 2 previous tests are sufficient)